src/unix_dgram.cc

```diff
@@ -224,20 +229,31 @@ NAN_METHOD(Bind) {
   sockaddr_un s;
   int err;
   int fd;
+  socklen_t namelen;
+  unsigned int len;
 
   assert(info.Length() == 2);
 
   fd = info[0]->Int32Value();
   String::Utf8Value path(info[1]);
 
+  len = path.length();
+  if (((*path)[0] == '\0' && len > sizeof(s.sun_path)) ||
+      ((*path)[0] != '\0' && len >= sizeof(s.sun_path))) {
+    err = -EINVAL;
+    goto out;
+  }
+  namelen = offsetof(struct sockaddr_un, sun_path) + len;
```

Comment:
There's already an `offset_of` macro defined in the file. I don't know which should be used. @bnoordhuis ?